provider/google.go

```diff
@@ -267,16 +254,58 @@ func (p *googleProvider) submitChange(zone string, change *dns.Change) error {
 		return nil
 	}
 
-	_, err := p.changesClient.Create(p.project, zone, change).Do()
+	zones, err := p.Zones()
 	if err != nil {
-		if !isNotFound(err) {
-			return err
+		return err
+	}
+
+	// separate into per-zone change sets to be passed to the API.
+	changes := separateChange(zones, change)
+
+	for z, c := range changes {
+		if _, err := p.changesClient.Create(p.project, z, c).Do(); err != nil {
```

Comment:
nitpick: I would name the funciton `changesByZone` to be more explicit what the function does